subprojects/core/src/main/java/org/gradle/caching/BuildCacheServiceFactory.java

```diff
@@ -40,14 +42,61 @@
  *     }
  * </pre>
  * </p>
- * @param <T> the type of build cache configuration this factory can handle.
  *
+ * @param <T> the type of build cache configuration this factory can handle.
  * @since 3.5
  */
 @Incubating
-public interface BuildCacheServiceFactory<T extends org.gradle.caching.configuration.BuildCache> {
+public interface BuildCacheServiceFactory<T extends BuildCache> {
+
     /**
-     * Creates a build cache service with the given configuration.
+     * Creates a build cache service from the given configuration.
+     *
+     * Implementations should also provide a description via the given describer.
      */
-    BuildCacheService createBuildCacheService(T configuration);
+    BuildCacheService createBuildCacheService(T configuration, Describer describer);
+
+    /**
+     * Builder-style object that allows build cache service factories to describe the cache service.
+     * <p>
+     * The description is for human consumption.
+     * It may be logged and displayed by tooling.
+     */
+    interface Describer {
+
+        /**
+         * Sets the description of the type of cache being used.
+         * <p>
+         * The value should not include particulars about the cache; only a human friendly description of the kind of cache.
+         * For example, instead of {@code "HTTP @ https://some/cache"} it should be just {@code "HTTP"}.
+         * Particular configuration should be set via {@link #config(String, String)}.
+         * <p>
+         * {@link BuildCacheServiceFactory} implementations should always return the same value for the same cache “type”.
+         * All implementations should call this method.
+         * <p>
+         * Values should be lowercase, except where using an acronym (e.g. HTTP).
+         * <p>
+         * Subsequent calls to this method replace the previously set value.
+         */
+        Describer type(String type);
+
+        /**
+         * Sets a configuration param of the cache being used.
+         * <p>
+         * e.g. {@code config("location", "https://some/cache")}.
+         * <p>
+         * Values may be logged.
+         * Secrets (e.g. passwords) should not be declared with this method.
+         * <p>
+         * Describing config is not required.
+         * Implementations should describe their config where possible.
+         * <p>
+         * Subsequent calls to this method with the same {@code name} argument will replace
+         * the previously supplied {@code value} argument.
+         * <p>
+         * Subsequent calls to this method with different {@code name} arguments will append values.
+         */
+        Describer config(String name, @Nullable String value);
```

Comment:
I just changed it to make both name/value required. I think this is simpler and better.